File: Dev Priya Gupta/Array/1313. Decompress Run-Length Encoded List.cpp
class Solution {
public:
    vector<int> decompressRLElist(vector<int>& nums) {
        int n=nums.size();
        vector<int> ans;
        for(int i=0;i<n;i+=2)  /// Time Complexity : O(n/2)
        {
            for(int j=1;j<=nums[i];j++)   /// Time Complexity in worst case O(100)
                ans.push_back(nums[i+1]);
        }
        return ans;
    }
};

Time Complexity : 
Space Complexity : O(n)

File: Vishakha Mishra /ARRAY/1313. Decompress Run-Length Encoded List.java
class Solution {
    public int[] decompressRLElist(int[] nums) {
        int n = nums.length;
       ArrayList<Integer> li = new ArrayList<>();
        
        for(int i = 1 ; i<n; i+=2){
            while(nums[i-1]>0){
                li.add(nums[i]);
                nums[i-1]--;
            }
        }
        int [] ans= new int[li.size()];
        for (int i = 0 ; i< li.size() ; i++){
            ans[i]= li.get(i);
            
        }
   return ans;
    }
}
